4.0.0 Changes - WIP

cs-abhishek-shukla opened the pull request.

cs-abhishek-shukla committed:
4.0.0 Changes - WIP

cs-ankit-gupta commented on release_notes.md:
```diff
@@ -1,18 +1,39 @@
+
+FortiAI intelligently selects appropriate Jinja expressions for each playbook step, reducing the need for manual adjustments.
+
+Structured outputs are now supported, with JSON data displayed in a readable JSON editor.
```
Structured outputs are now supported,  


This is part of OpenAI connector.

cs-abhishek-shukla replied:
![image](https://github.com/user-attachments/assets/60717953-66d9-4f68-9625-82242da9df03)

cs-ankit-gupta commented on release_notes.md:
```diff
@@ -1,18 +1,39 @@
-Now, threads are retained within OpenAI, enabling improved historical reference and continuity. Each new prompt, after clearing the conversation. still initiates a fresh thread, and the *Clear Conversation* option within the user interface continues to clear conversation history, ensuring a clean workspace.
\ No newline at end of file
+- Playbook creation from simple prompts or clicks
+- Accurate detection of connector step types, operation names, and available inputs
```
Can we rephrase it to like with some additional content as

Correct selection of playbook step type along with module name, connector and operation name.

cs-ankit-gupta commented on release_notes.md:
```diff
@@ -1,18 +1,39 @@
+    - Insufficient permissions
+    - Failure during assistant creation or update
+    - Missing default connector configuration in single-user setups
+    - Unspecified user configuration in multi-user environments
```
All this were already present. we are not changing any thing in this.

cs-abhishek-shukla committed:
4.0.0 Changes - WIP

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
+- Accurate detection of connector step types, operation names, and available inputs
+- Identification of steps that require loops for record traversal
+
+### Improved Jinja Expression Selection
```
This is also a part of  "Key improvements"

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
+FortiAI now enables connector creation using natural language. By providing inputs such as CURL commands, API details, and configuration parameters, users can generate a fully functional connector that includes Connector configuration, Auto-generated connector code, and Sample playbooks.
 
-### Voice Recognition for Hands-Free Interaction  
+You can also upload attachments to assist in connector generation.
```
please add supported attachment type like postman collection, swagger, etc.

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
+The configuration wizard has been improved to support the following:
 
-Previously, conversation threads were deleted from OpenAI whenever *Clear Conversation* was clicked, and a new thread was started with each new prompt.
+- Creation of SOC and Playbook assistants
```
Creation of SOC and Playbook assistants --> this was already present. Connetor assistant is new this time

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
 
-Previously, conversation threads were deleted from OpenAI whenever *Clear Conversation* was clicked, and a new thread was started with each new prompt.
+- Creation of SOC and Playbook assistants
+- Updating assistant instruction sets
```
this is also present from the previous release

cs-suyog-jinturkar replied:
same for below points as well

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
+
+- **Structured outputs**: JSON data is now displayed in a JSON editor for improved readability.
+
+- **Clear conversation**: The "Clear Conversation" feature is now available for both playbook and connector generation. It also clears conversation threads and any attached temporary files.
```
any attached temporary files from FortiSOAR and OpenAI

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
+
+- **Context-aware assistant loading**: The bot now detects the active page and loads the appropriate assistant automatically.  
+  - On the Content Hub "Create" page, the Connector Generation Assistant will be available.  
+  - On the Playbook Designer page, the Playbook Assistant will be loaded.
```
Please also mention about SOC assistant as well, that it will be available on modules like Alerts, Incidents, etc.

cs-abhishek-shukla replied:
SOC assistant was always present and hence cannot be part of "What's new".

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
+- **Load past conversations**:  
+  - Users can now see past conversations relevant to the specific page they are on.  
+    - On the Connector Generation page (Content Hub - Create), only previous connector-related messages will be shown.  
+    - On the Playbook Designer page, only playbook-related messages will be displayed.
```
same here as well, please add SOC assistant related details here as well

cs-abhishek-shukla replied:
SOC assistant was always present and hence cannot be part of "What's new".

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,18 +1,55 @@
+    - On the Playbook Designer page, only playbook-related messages will be displayed.
+
+> [!Note]
+> Past conversations loads after logout and login, only for the SOC Assistant.
```
this note is not valid anymore, please rephrase it as, past conversation loads after logout and login, for all assistants like SOC Assistant, Playbook and Connector

cs-suyog-jinturkar commented on README.md:
```diff
@@ -6,8 +6,8 @@
+> [!NOTE]
+> This is a preview version of **FortiAI** designed to demonstrate the potential of Generative AI in FortiSOAR. Preview releases are a beta release. This means that release is intended to get feedback and might not be best suited for production level deployments. Feature availability and support for preview releases will continue to improve as the solution/feature matures.
 
 # Overview
```
Overview should also now talk about connector generation

cs-suyog-jinturkar replied:
In overview, like we mentioned about "Filtering Feature Overview" we should now also add about advance PB generation and connector generation before "Filtering Feature Overview"

cs-suyog-jinturkar commented on README.md:
```diff
@@ -48,6 +48,8 @@ For any further questions or concerns about your privacy, refer to [OpenAI’s p
     - [Prompts for filtering *alert* records](./docs/prompts-for-filtering-alert-records.md)
     - [Prompts for filtering *incident* records](./docs/prompts-for-filtering-incident-records.md)
     - [Prompts for filtering *indicator* records](./docs/prompts-for-filtering-indicator-records.md)
+    - [Prompts for creating a connector](./docs/prompts-for-creating-connectors.md)
```
for PB generation?

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -59,20 +72,20 @@ The **FortiAI** solution pack contains the following resources:
+| Get Playbook Step Suggestion (Loop) ![](./res/icon-deprecated.svg) | Suggests steps to create a playbook based on user's query, in a loop.                     |
+| Get Playbook Block Suggestion ![](./res/icon-deprecated.svg)       | Creates a suggested playbook block to perform actions based on user requirements.         |
+| > Get LLM Response ![](./res/icon-deprecated.svg)                  | Reference playbook for getting a response from the LLM.                                   |
+| > Get LLM Configuration ![](./res/icon-deprecated.svg)             | Get the LLM configuration based on the value defined for the key "isMultiConfigAvailable" |
```
Please add newly added PB "Clear Assistant metadata" here. @cs-ankit-gupta  can give more details here

cs-suyog-jinturkar commented on docs/prompts-for-creating-connectors.md:
```diff
@@ -0,0 +1,37 @@
+
+* **Prompt**:
+
+>   1. _Build connector for Criminal IP product for following actions:_
```
We will need more example prompts, their step by step user interaction along with screenshots and end result in create tab and in BYOC (BYOC details are good for only one example for rest we can skip) 
@amol can help here

cs-suyog-jinturkar commented on docs/prompts-for-creating-connectors.md:
```diff
@@ -0,0 +1,37 @@
+
+>   - Get Domain Reputation "_
+
+* **Expected Outcome**: Displays only alerts whose *Status* is **Investigating**.
```
This is not needed

cs-suyog-jinturkar commented on docs/prompts-for-creating-connectors.md:
```diff
@@ -0,0 +1,37 @@
+
+* **Expected Outcome**: Displays only alerts whose *Status* is **Investigating**.
+
+![Under investigation alerts filtered](./res/filter-alert-investigating.png)
```
This is not needed

cs-suyog-jinturkar commented on docs/prompts-for-creating-connectors.md:
```diff
@@ -0,0 +1,37 @@
+
+#### Review results
+
+Verify that the results include only alerts marked as **Investigating**, ensuring that irrelevant alerts are excluded as per the filtering criteria.
```
This should be corrected with connector generation details

cs-suyog-jinturkar commented on docs/troubleshooting.md:
```diff
@@ -147,9 +147,83 @@ In some cases, the field names in filters may not reflect changes after subseque
 
+> [!Note]
+> This issue has been resolved in **FortiSOAR v7.6.1** and later.
 
```
Troubleshooting should have clear three section for SOC, PB and Connector generatio

cs-abhishek-shukla replied:
For the users it is one assistant and hence creating separate sections is redundancy, IMO.

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -5,8 +5,9 @@
 # Usage
 
-The **FortiAI** solution pack offers a variety of functions through its **AI Assistant** to managing and automating tasks related to cybersecurity workflows. Following is an overview of the supported functions:
+The **FortiAI** solution pack offers a variety of functions for managing and automating tasks related to cybersecurity workflows. Following is an overview of the supported functions:
```
variety of functions or variety of feature?

cs-abhishek-shukla replied:
Functions. Not in the code sense. In general, non-techy, users' perspective

cs-ankit-gupta commented on docs/contents.md:
```diff
@@ -36,16 +36,29 @@ The **FortiAI** solution pack contains the following resources:
 
 ### `fortiai-configurations`
 
+{
```
Can we directly add link here as 'docs/llm-configuration.json'

cs-ankit-gupta commented on docs/setup.md:
```diff
@@ -84,8 +90,6 @@ The **FortiAI** solution depends on the following connectors and widgets:
 >[!NOTE]
```
To be added on line 50: 
Delete permission for file

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -5,8 +5,9 @@
-The **FortiAI** solution pack offers a variety of functions through its **AI Assistant** to managing and automating tasks related to cybersecurity workflows. Following is an overview of the supported functions:
+The **FortiAI** solution pack offers a variety of functions for managing and automating tasks related to cybersecurity workflows. Following is an overview of the supported functions:
 
+- **Create Connector**: Create a connector using natural language prompts, review results, and download the associated connector code in the form of `info.json`, `operations.py` etc.
```
download the is not supported

cs-suyog-jinturkar replied:
FortiAI will import that conenctor into FSR

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -5,8 +5,9 @@
+- **Create Connector**: Create a connector using natural language prompts, review results, and download the associated connector code in the form of `info.json`, `operations.py` etc.
 - **Generate Playbook**: Generate, create a playbook or workflow, or playbook steps.
 - **Navigate Pages**: Navigate, open, or move to a specific page or module.
 - **Link Record**: Link or correlate a record with other relevant records. For example: correlating an indicator to an existing alert or incident.
```
We also do Alert Triage as part of the SOC assistant

cs-abhishek-shukla replied:
SOC assistant, Playbook assistant, Connector assistant are all distinctive internally, There is only ONE assistant. AI Assistant/Chatbot.

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -22,6 +23,40 @@ The **FortiAI** solution pack offers a variety of functions through its **AI Ass
+   API details can be provided in various formats, including:
+   - `curl` commands  
+   - JSON request/response structures  
+   - File attachments in formats such as `.json`, `.yml`, `.txt`, or `.pdf`
```
Please add a Note: attachement size should not be greater than 100 MB

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -22,6 +23,40 @@ The **FortiAI** solution pack offers a variety of functions through its **AI Ass
+
+2. **Code Generation**  
+   Based on the provided input, FortiAI will automatically generate the standard connector code files:
+   - `info.json`  
```
this is a metadata file and not a code file

cs-suyog-jinturkar replied:
requirments.txt will also be generated if there are any 3rd prty dependencies are discovered in the code

cs-suyog-jinturkar replied:
This will also generate sample playbooks for respective conenctor actions.

cs-avit-suryawanshi commented on docs/contents.md:
```diff
@@ -36,16 +36,29 @@ The **FortiAI** solution pack contains the following resources:
-    - `pBGenerationModel`: Sets the model of LLM Integration to use for playbook generation. By Default, `gpt-4o-mini` is to be used.
-    - `isMultiConfigAvailable`: Sets the availability of multiple configuration. By Default, it is set to `false`.
+- `isMultiConfigAvailable`: Sets the availability of multiple configuration. By Default, it is set to `false`.
+- `pBGenerationModel`: Sets the model of LLM Integration to use for playbook generation. By Default, `gpt-4o-mini` is to be used.
```
Remove "pBGenerationModel"  as we have removed it in this version.
Add - 
"pastConversationMsgLimit"  : Specify the maximum messages to be loaded in the conversation window on login. By default, it is set to `20`.

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -22,6 +23,40 @@ The **FortiAI** solution pack offers a variety of functions through its **AI Ass
+   - `operations.py`  
+
+3. **Output and Packaging**  
+   The generated files, along with sample playbooks, are bundled and the connector is automatically imported into the **Build Your Own Connector (BYOC)** section within FortiSOAR.
```
connector will be imported into create tab (user will need to refersh it) to see it.
From there, user can edit it into BYOC to edit and review the connector metadata and code

cs-avit-suryawanshi commented on docs/setup.md:
```diff
@@ -20,17 +20,23 @@ The **FortiAI** solution pack depends on the following solution packs.
 
 - An OpenAI account and a valid project level API key to access the OpenAI APIs. For information refer to [API Keys](https://platform.openai.com/docs/api-reference/api-keys) section of the OpenAI documentation. Following are the **_minimum_** permissions that must be assigned to users on the OpenAI interface:
+
     - **Administrator**
```
Update the Permissions :
For Administrator
    Model - Read
    Model Capabilites - Write
    Assistants - Write
    Threads - Write
    Files - Write

For User
For Administrator
    Model - Read
    Model Capabilites - Write
    Assistants - Read
    Threads - Write
    Files - Write

cs-abhishek-shukla committed:
4.0.0 Changes - WIP
- Incorporated feedback comments
- Rewrote the README.md
- Rewrote the release notes
- Edited contents.md
- Work on progress

cs-amol-phatangade commented on README.md:
```diff
@@ -6,39 +6,70 @@
+FortiAI introduces a game-changing **Connector Generation** feature, allowing users to create custom connectors through **natural language commands**. Whether you need to integrate new tools, systems, or APIs, FortiAI takes care of the heavy lifting:
+
+- **Auto-Generated Connector Code**: With a few simple inputs, you get fully functional connectors ready to integrate into your workflows.
+- **Seamless Configuration**: Easily configure connectors using CURL commands, API details, and additional parameters.
```
Need to rephrase this sentence.

cs-abhishek-shukla replied:
Removed

cs-abhishek-shukla committed:
4.0.0 Changes
- Added connector generation doc
- PB Generation doc is WIP

cs-abhishek-shukla committed:
4.0.0 Changes:
- Playbook generation

cs-abhishek-shukla committed:
4.0.0 Changes:
- Removed json doc

cs-avit-suryawanshi commented on docs/troubleshooting.md:
```diff
@@ -147,9 +147,83 @@ In some cases, the field names in filters may not reflect changes after subseque
+5. Save the changes.
+
+> [!NOTE]
+> You may need admin permissions on the project to modify rate limits.
```
Add - 
Indicators Not Unmasked in Playbook Steps on Designer Page
Detail:
If the playbook outline contains indicators, they remain masked and are not automatically unmasked in the generated step.

Resolution:
Manually update the masked indicators with their actual (unmasked) values.

cs-abhishek-shukla committed:
4.0.0 Changes
- Added a deferred bug to troubleshooting
